### ISSUE TITLE
Another potential switch issue?

### ISSUE THREAD

**Issue by andreasfaerber** [dworkin/dgd#91]
Greetings,

having recently updated to DGD 1.7.5 we encountered some switch issue which seems similiar to the issues already discussed and fixed in past issues. For certain switches it will not match the 0 based on ranged used in the switch statement. Example code:

```
void switch_test() {
    int val;

    val = 0;

    write("First\n");
    switch(val) {
      case -900..-500:
          write("-900..-500\n");
          break;
      case -499..-100:
          write("-499..-100\n");
          break;
      case -99..-20:
          write("-99..-20\n");
          break;
      case -19..-1:
          write("-19..-1\n");
          break;
      case 0..20:
          write("0..20\n");
          break;
      case 21..50:
          write("21..50\n");
          break;
      case 51..100:
          write("51..100\n");
          break;
      case 101..200:
          write("101..200\n");
          break;
      case 201..500:
          write("201..500\n");
          break;
      case 501..4000:
          write("501..4000\n");
          break;
      default:
          write("default\n");
    }
    write("Second\n");
    switch(val) {
      case -900..-500:
          write("-900..-500\n");
          break;
      case -499..-1:
          write("-499..-1\n");
          break;
      case 0..20:
          write("0..20\n");
          break;
      case 21..50:
          write("21..50\n");
          break;
      case 51..100:
          write("51..100\n");
          break;
      case 101..200:
          write("101..200\n");
          break;
      case 201..500:
          write("201..500\n");
          break;
      case 501..4000:
          write("501..4000\n");
          break;
      default:
          write("default\n");
    }
}
```
Calling the switch_test() function will result in the following output:

```
First
default
Second
0..20
```

Hm. Am i missing something or may this be another switch weirdness?

Regards
Andreas / Tarjan@Xyllomer

**Comment by andreasfaerber** [comment]
Some more testing: Any negative val(ue) results both switch statements to default. Anything above 20 seems to work for both.

**Comment by dworkin** [comment]
I will take a look at this.

**Comment by dworkin** [comment]
Should be fixed by 85f8d807c48dd2334897338e8c237d9bb3e7eb53.

**Comment by andreasfaerber** [comment]
Looks good and fixed to me!
